src/polyfill.rs

```diff
@@ -178,3 +178,20 @@ macro_rules! slice_as_array_ref_mut {
         }
     }
 }
+
+/// Converts a boolean into a `Result<(), err::Unspecified>` in the
+/// obvious way. By composing this with `try!`, we can compactly
+/// represent `Result`-valued early returns that guard on a condition.
+macro_rules! check {
```

Comment:
> This doesn't need to be a macro, does it? I think it can be an inline function.

It doesn't— it only started life as a macro so I could do an early return, then later realized it made more sense to just compose it with `try!`.

Looking at it again, I don't think `polyfill` is the best module for this. Maybe `error`, instead?

> IMO it would be better to do this as a separate PR

Yes, that makes sense to me. Will try that tonight.